src/Tus/Client.php

```diff
@@ -426,19 +426,44 @@ public function getOffset()
      */
     public function create(string $key) : string
     {
+        return $this->createWithUpload($key, 0)['location'];
+    }
+
+    /**
+     * Create resource with POST request and upload data using the creation-with-upload extension
+     *
+     * @see https://tus.io/protocols/resumable-upload.html#creation-with-upload
+     *
+     * @param string $key
+     * @param int $bytes -1 => all data; 0 => no data
+     * @return array ['location' => $uploadLocation, 'offset' => $offset]
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function createWithUpload(string $key, int $bytes = -1) : array
```

Comment:
> -1 => all data; 0 => no data

In the implementation below we always check if `$bytes > 0` to decide if we need to send data to the server to upload. In that case, doesn't it make more sense to use zero as a default value for `$bytes`?

Reply:
It's to be consistent with the `upload` function.
And to use that function with `$bytes = 0` does not make any sense, then you would use the `create` function. The option to send `$bytes = 0` is really only there to reuse the code also for the `upload` function.

It's your call, whatever you prefer more.

Reply:
I got your point now, negative default value is fine. Thanks!